pkg/volume/quobyte/quobyte.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package quobyte
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/util/strings"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// ProbeVolumePlugins is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{&quobytePlugin{nil}}
+}
+
+type quobytePlugin struct {
+	host volume.VolumeHost
+}
+
+var _ volume.VolumePlugin = &quobytePlugin{}
+var _ volume.PersistentVolumePlugin = &quobytePlugin{}
+
+const (
+	quobytePluginName = "kubernetes.io/quobyte"
+)
+
+func (plugin *quobytePlugin) Init(host volume.VolumeHost) error {
+	plugin.host = host
+	return nil
+}
+
+func (plugin *quobytePlugin) GetPluginName() string {
+	return quobytePluginName
+}
+
+func (plugin *quobytePlugin) GetVolumeName(spec *volume.Spec) (string, error) {
+	volumeSource, _, err := getVolumeSource(spec)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf(
+		"%v:%v",
+		volumeSource.Registry,
+		volumeSource.Volume), nil
+}
+
+func (plugin *quobytePlugin) CanSupport(spec *volume.Spec) bool {
+	if (spec.PersistentVolume != nil && spec.PersistentVolume.Spec.Quobyte == nil) ||
+		(spec.Volume != nil && spec.Volume.Quobyte == nil) {
+		return false
+	}
+
+	// If Quobyte is already mounted we don't need to check if the binary is installed
+	if mounter, err := plugin.newMounterInternal(spec, nil, plugin.host.GetMounter()); err == nil {
+		qm, _ := mounter.(*quobyteMounter)
+		pluginDir := plugin.host.GetPluginDir(strings.EscapeQualifiedNameForDisk(quobytePluginName))
+		if mounted, err := qm.pluginDirIsMounted(pluginDir); mounted && err == nil {
+			glog.V(4).Infof("quobyte: can support")
+			return true
+		}
+	} else {
+		glog.V(4).Infof("quobyte: Error: %v", err)
+	}
```

Comment:
We check if Quobyte is already mounted if this is true we don't need to check for the binary (which allows us to use the plugin also on CoreOS, Rancher etc). AFAIK there is no containerized mount in Kubernetes?